langchain/schema.py

```diff
@@ -339,12 +339,21 @@ async def aget_relevant_documents(self, query: str) -> List[Document]:
 T = TypeVar("T")
 
 
-class BaseOutputParser(Serializable, ABC, Generic[T]):
+class BaseLLMOutputParser(Serializable, ABC, Generic[T]):
+    @abstractmethod
+    def parse_result(self, result: List[Generation]) -> T:
+        """Parse LLM Result."""
```

Comment:
nit: this is not an LLMResult but a list of generations (we have an LLMResult model so naming is introducing collisions)